polymer/src/sample-category-data.html

```diff
@@ -0,0 +1,203 @@
+<!--
+@license
+Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+-->
+
+<link rel="import" href="../node_modules/@npm-polymer/polymer/polymer-element.html">
+<link rel="import" href="../node_modules/@npm-polymer/iron-ajax/iron-ajax.html">
+
+<dom-module id="sample-category-data">
+  <script>
+
+
+  (function() {
+
+    class SampleCategoryData extends Polymer.Element {
+
+      static get is() { return 'sample-category-data'; }
+
+      static get properties() { return {
+
+        categoryName: String,
+
+        itemName: String,
+
+        categories: {
+          type: Array,
+          notify: true
+        },
+
+        category: {
+          type: Object,
+          //computed: '_computeCategory(categoryName)',
+          notify: true
+        },
+
+        item: {
+          type: Object,
+          computed: '_computeItem(category.items, itemName)',
+          notify: true
+        },
+
+        failure: {
+          type: Boolean,
+          notify: true,
+          readOnly: true
+        }
+
+      }}
+
+      static get observers() { return [
+        '_categoriesLoaded(categories)', '_categoryNameSet(categoryName)'
+      ]}
+
+      _categoriesLoaded(categories) {
+        this._computeCategory(this.categoryName)
+      }
+
+      _categoryNameSet(categoryName) {
+        this._fetchCategories(3);
+      }
+
+
+      _getCategoryObject(categoryName) {
+        if (this.categories === undefined) {
+          return null;
+        }
+        for (let i = 0, c; c = this.categories[i]; ++i) {
+          if (c.id === categoryName) {
+            this.category = c;
+            return c;
+          } else if (categoryName === "index.html") {
+            return null
+          }
+        }
+      }
+
+      _computeCategory(categoryName) {
+
+        // Fetch the items of the category. Note that the fetch is asynchronous,
+        // which means `category.items` may not be set initially (but that path
+        // will be notified when the fetch completes).
+        let categoryObj = this._getCategoryObject(categoryName);
+        this._fetchItems(categoryObj, 1);
+        return categoryObj;
+      }
+
+      _computeItem(items, itemName) {
+        if (!items || !itemName) {
+          return;
+        }
+        for (let i = 0, item; item = items[i]; ++i) {
+          if (item.url.endsWith(itemName.concat('/'))) {
+            return item;
+          }
+        }
+      }
+
+      handleResponse(data) {
```

Comment:
I think this is duplicate from the app element.

Reply:
Yep, removed